### ISSUE TITLE
Description of "legend_loc" option in scanpy.pl.embedding is inaccurate or not implemented yet.

### ISSUE THREAD

**Issue by adkinsrs** [scverse/scanpy#2229]
- [X] I have checked that this issue has not already been reported.
- [X] I have confirmed this bug exists on the latest version of scanpy.
- [X] (optional) I have confirmed this bug exists on the master branch of scanpy.

---

**Note**: Please read [this guide](https://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports) detailing how to provide the necessary information for us to reproduce your bug.

In the documentation at https://scanpy.readthedocs.io/en/stable/generated/scanpy.pl.embedding.html#scanpy.pl.embedding, the description of the "legend_loc" param is as follows:

> legend_loc : [str](https://docs.python.org/3/library/stdtypes.html#str) (default: 'right margin')
> Location of legend, either 'on data', 'right margin' **or a valid keyword for the loc parameter of [Legend](https://matplotlib.org/stable/api/legend_api.html#matplotlib.legend.Legend).**

The part in bold does not seem to be implemented, as the "legend_loc" only has if/elif conditions for it is "right margin" or "on data"... nothing for the matplotlib "loc" keywords.  I am not sure if it is best to implement this to match the description, or change the description to omit the part in bold.

### Minimal code sample (that we can copy&paste without having any data)
https://github.com/scverse/scanpy/blob/2e98705347ea484c36caa9ba10de1987b09081bf/scanpy/plotting/_tools/scatterplots.py#L55-L461

https://github.com/scverse/scanpy/blob/2e98705347ea484c36caa9ba10de1987b09081bf/scanpy/plotting/_tools/scatterplots.py#L1099-L1109

#### Versions

Not applicable

**Comment by LisaSikkema** [comment]
Yeah agree this is inconsistent, I'd say we should either correct the documentation + throw an error when the wrong argument is passed, or implement the matplotlib options. @ivirshup what do you think?

In [_anndata.py](https://github.com/scverse/scanpy/blob/11d0b8e992ad145eeb3f666aa4e006bd204272de/scanpy/plotting/_anndata.py#L38-L54)  plotting script, VALID_LEGENDLOCS is set (not sure in which plotting functions that is eventually used), could we use something similar for the above example?
```python
VALID_LEGENDLOCS = {
    'none',
    'right margin',
    'on data',
    'on data export',
    'best',
    'upper right',
    'upper left',
    'lower left',
    'lower right',
    'right',
    'center left',
    'center right',
    'lower center',
    'upper center',
    'center',
}
```
Also related to issue #2322

**Comment by LisaSikkema** [comment]
Oh this might actually be addressed in #2267
